keyboards/handwired/bongo80/keymaps/default/keymap.c

```diff
@@ -0,0 +1,179 @@
+};
+
+#ifdef OLED_ENABLE
+
+// Time per frame in millis
+#define FRAME_LENGTH_SLOW 400
+#define FRAME_LENGTH_MED 300
+#define FRAME_LENGTH_FAST 200
+
+// Represents an animation, including the frames and length (time) per frame
+struct frame_set {
+	const int len;
+	int frame_len;
+	const char** frames;
+};
+
+enum state {
+	IDLE = 0, SLOW = 1, MED = 2, FAST = 3
+};
+
+// Framesets for each typing state
+struct frame_set no_caps[] = {
+	{.frames = idle_no_caps, .len = idle_len, .frame_len = FRAME_LENGTH_SLOW},
+	{.frames = slow_no_caps, .len = slow_len, .frame_len = FRAME_LENGTH_SLOW},
+	{.frames = med_no_caps, .len = med_len, .frame_len = FRAME_LENGTH_MED},
+	{.frames = fast_no_caps, .len = fast_len, .frame_len = FRAME_LENGTH_FAST}
+};
+
+struct frame_set caps[] = {
+	{.frames = idle_caps, .len = idle_len, .frame_len = FRAME_LENGTH_SLOW},
+	{.frames = slow_caps, .len = slow_len, .frame_len = FRAME_LENGTH_SLOW},
+	{.frames = med_caps, .len = med_len, .frame_len = FRAME_LENGTH_MED},
+	{.frames = fast_caps, .len = fast_len, .frame_len = FRAME_LENGTH_FAST}
+};
+
+struct frame_set* frame_sets[] = {no_caps, caps};
+struct frame_set* curr_frame_set = &no_caps[0];
+int curr_frame_index = 0;
+
+int curr_wpm = 0;
+led_t led_usb_state;
+bool run_oled = 1;
+uint32_t time = 0;
+
+static void render_animate(void) {
+
+	// Writes the WPM to the screen
+	char to_write[32];
+	if (led_usb_state.caps_lock) {
+		sprintf(to_write, "WPM:%d CAPS  ", curr_wpm);
+	} else {
+		sprintf(to_write, "WPM:%d       ", curr_wpm);
+	}
+
+	oled_set_cursor(0, 7);
+	oled_write(to_write, false);
+	oled_set_cursor(0, 0);
+
+	// Updates frame being displayed based on set interval
+    if (timer_elapsed(time) > curr_frame_set->frame_len) {
+
+        time = timer_read();
+
+		// Ensures a smoothly animated transition between the different states of the animation
+		if (curr_frame_index == curr_frame_set->len - 1) {
+
+			// Assigns frameset based on typing speed
+			int frame_set_index = 0;
+			if (curr_wpm <= 3) {
+				frame_set_index = IDLE;
+
+			} else if (curr_wpm <= 20) {
+				frame_set_index = SLOW;
+
+			} else if (curr_wpm <= 35) {
+				frame_set_index = MED;
+
+			} else {
+				frame_set_index = FAST;
+			}
+
+			curr_frame_set = &frame_sets[led_usb_state.caps_lock][frame_set_index];
+			curr_frame_index = 0;
+
+		} else {
+        	curr_frame_index++;
+		}
+
+		oled_write_raw_P(curr_frame_set->frames[curr_frame_index], frame_size);
+    }
+}
+
+bool process_record_kb(uint16_t keycode, keyrecord_t *record) {
+	switch (keycode) {
+		// Handles the keycode for turning on and off the oled screen
+		case KC_OLED_STATE:
+			if (record->event.pressed) {
+				run_oled = !run_oled;
+				oled_clear();
+			}
+			return false;
+	}
+	return true;
+}
+
+bool oled_task_user() {
+
+	curr_wpm = get_current_wpm();
+	led_usb_state = host_keyboard_led_state();
+	if (run_oled) {
+    	render_animate();
+	}
+    return false;
+}
+
+#endif
```

Comment:
This should be moved to the keyboard level in a `bongo80.c` file using the appropriate `_kb` functions.
```suggestion
```